### PR TITLE
Update contributor guide and add a known issue

### LINKED ISSUE #444: Add a known issue that volume expansion is not supported when there are snapshots
**Describe the bug**
If upload fails during backup, the local snapshot will not be deleted.  Velero vSphere Plugin will keep retrying the upload. If user wants to expand volume, it will fail because volume expansion is not supported when there are snapshots on the volume.

We should document this in the Known Issues section.

[A clear and concise description of what the bug is]

**To Reproduce**

[Steps to reproduce the behavior]

**Expected behavior**

[A clear and concise description of what you expected to happen]

**Troubleshooting Information**

[Please refer to the [Troubleshooting](https://github.com/vmware-tanzu/velero-plugin-for-vsphere/blob/main/docs/troubleshooting.md) page and collect the required information]

**Screenshots**

[If applicable, add screenshots to help explain your problem.]

**Anything else you would like to add:**

[Miscellaneous information that will assist in solving the issue]

### PR DESCRIPTION
**Pull request by xing-yang**
**What this PR does / why we need it**:
Update the contributor guide and add a known issue that volume expansion is not supported when there are snapshots on the volume.

**Which issue(s) this PR fixes**:

Fixes #444 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/678/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/679/

File: docs/known_issues.md [hunk 1]
```diff
@@ -60,3 +60,7 @@ velero backup create my-backup --include-namespaces my-namespace --snapshot-volu
 Users might require ways to provide registry credentials when deploying velero and vSphere plugin with images in private registries. Below are solutions in different cases.
 * In Vanilla cluster or Tanzu Kubernetes Grid Service cluster, users can [add the image pull secrets to the corresponding service accounts](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#add-image-pull-secret-to-service-account).
 * In vSphere with Tanzu cluster, users can use the `--use-private-registry` option available in [velero vSphere operator CLI](velero-vsphere-operator-cli.md) to indicate pulling image from private registry. As a prerequisite, users are expected to provide the registry credential when enabling `Velero vSphere Operator` Supervisor Service.
+
+## Volume Expansion Not Supported when there are Snapshots
+
+If upload fails during a backup, the local snapshot will not be deleted. Velero vSphere Plugin will keep retrying the upload. If the user wants to expand volume, it will fail because volume expansion is not supported when there are snapshots on the volume.
```
Comment (deepakkinni) [inline docs/known_issues.md:66]: Do we want to recommend some solutions here?, like deleting the FCD snapshot from mob?. In the future release, they can even create the DeleteSnapshot CR.

Comment (xing-yang) [reply docs/known_issues.md:66]: Discussed about this.  We'll add the work around when the DeleteSnapshot CR approach becomes available in the next release.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,15 +1,7 @@
...
-Authors are expected to include a changelog file with their pull requests. The changelog file
-should be a new file created in the `changelogs/unreleased` folder. The file should follow the
-naming convention of `pr-username` and the contents of the file should be your text for the
-changelog.
-
-    velero-plugin-for-vsphere/changelogs/unreleased   <- folder
-        000-username            <- file
-
+## Opening PRs
+When opening a pull request, please fill out the checklist supplied the template. This will help others properly categorize and review your pull request.
```
Comment (lintongj) [inline CONTRIBUTING.md:4]: Do we want to explicitly emphasize that release-note should be filled in?

File: docs/known_issues.md (hunk 1 above)
Comment (lintongj) [reply docs/known_issues.md:66]: BTW, we should also mention: before applying any workaround to delete snapshot, no matter whether it is FCD DeleteSnapshot API or DeleteSnapshot CRD, user should make sure cancel the backup if upload is still retrying.

Comment (xing-yang) [reply docs/known_issues.md:66]: Good point.  We should mention that when we provide a workaround.

File: CONTRIBUTING.md (hunk 2 above)
Comment (xing-yang) [reply CONTRIBUTING.md]: Added.

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/682/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/2/

**Review by lintongj — APPROVED** [review]

**Pull request merged**